### ISSUE TITLE
Issue with Android device connectivity when connected to custom Access Point

### ISSUE THREAD

**Issue by jchri853** [falconeta/capacitor-wifi-connect#16]
I'm currently working on a project where I need to connect my mobile device (both iOS and Android) to a custom Access Point (AP) on a hardware device and then perform API calls to retrieve information from the device. While this setup works smoothly on iOS, I'm encountering difficulties with Android.

After successfully connecting my Android device to the custom AP using my app, the device indicates that it's not connected to any network, but it connect via the app, causing my API calls to fail. This behavior differs from iOS, where the connection to the AP on an OS level? and is established without issues.

How does Android handle connections to custom Access Points compared to iOS?
Is there a known workaround or solution to ensure proper network connectivity on Android devices when connected to a custom AP via an app?

* I've verified that the Android device is connected to the AP through the app's interface.
* The API calls work as expected on iOS devices under the same conditions.
* I'm using native fetch methods for API calls within my app.

Any insights or suggestions would be greatly appreciated. Thanks in advance for your help! This plugin is fantastic for our iOS users.

**Comment by falconeta** [comment]
Hi @jchri853! Strange behaviour, can you share some details about Android device? (es. model, OS version).

**Comment by jchri853** [comment]
Galaxy S9 SM-G960U Android 10
Pixel 4a (5G)  Android 14 
Galaxy A53 5G Android 14



![network](https://github.com/falconeta/capacitor-wifi-connect/assets/3412424/f50faf91-b393-432c-91f6-be70e5606ddd)

**Comment by falconeta** [comment]
Hi @jchri853, unfortunately I don't reproduce the issue with my example device (simple esp32 with HTTP server).

This line prepare the request to access at AP:
https://github.com/falconeta/capacitor-wifi-connect/blob/52dd1ef96d18d9bc54660bb35336887cbfc92f54/android/src/main/kotlin/com/falconeta/capacitor_wifi_connect/CapacitorWifiConnect.kt#L458C5-L462C15

Here the google DOC: 
https://developer.android.com/reference/android/net/NetworkRequest.Builder

Try modifying the plugin code by adding or removing the capabilities that might trigger the problem

**Comment by jchri853** [comment]
Thank you, @falconeta, for the invaluable tips. Upon further experimentation with capabilities, I discovered that my web application reported being "offline" while being connected to an access point via the plugin on Android.

I found using https://capacitorjs.com/docs/apis/http to make native http calls to my access point was the solution over the cordova http plugin i had been using.

Again thanks for this great plugin, this will greatly improve my customers provisioning experience.
